File: Selenium/Activities/Activity12.java
import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity12 {
	public static void main(String args[]) {
		WebDriver driver = new FirefoxDriver();
		WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(30));
		driver.get("https://training-support.net/webelements/dynamic-content");
		System.out.println("The title of the page is: "+driver.getTitle());
		
		WebElement button =driver.findElement(By.id("genButton"));
		button.click();
		
		if(wait.until(ExpectedConditions.textToBePresentInElementLocated(By.id("word"),"release"))) {
			WebElement word = driver.findElement(By.id("word"));
			System.out.println("Word is: "+word.getText());
		}
		
		driver.quit();
	}

}

File: Selenium/Activities/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
with webdriver.Firefox() as driver:
    wait = WebDriverWait(driver,timeout=10)
    driver.get("https://training-support.net/webelements/dynamic-content")
    print(f"the title of the page is {driver.title}")

    button =driver.find_element(By.ID,"genButton")
    button.click()

    if wait.until(EC.text_to_be_present_in_element((By.ID,"word"),"release")):
        word = driver.find_element(By.ID,"word")
        print("The word is: ",word.text)

    driver.quit()
